File: python-pytest/abcmouse/base/page.py
import time
from abc import ABC


class Page(ABC):
    def __init__(self, driver, url=None):
        self.driver = driver
        self.url = url

    def js_locator(self, locator_parts):
        return self.driver.execute_script("return " + "".join(locator_parts))

    def wait(self, seconds):
        time.sleep(seconds)

    def navigate_to(self):
        self.driver.get(self.url)
        self.wait(3)

File: CSharpNunitSelenium/CSharpNunitSelenium/ABCMouse/Base/Page.cs
using OpenQA.Selenium.Support.UI;

namespace CSharpNunitSelenium.ABCMouse.Base;

using OpenQA.Selenium;

public abstract class Page : IPage
{
    protected IWebDriver Driver { get; }
    protected readonly IJavaScriptExecutor? _jsExecutor;
    protected abstract string URL { get; set; }

    protected Page(IWebDriver driver)
    {
        Driver = driver;
        _jsExecutor = (IJavaScriptExecutor) driver;
    }
    
    public void NavigateTo()
    {
        Driver.Navigate().GoToUrl(URL);
    }
    
    public void Wait(double seconds)
    {
        Thread.Sleep(TimeSpan.FromSeconds(seconds));
    }

    public string GetUrl()
    {
        return Driver.Url;
    }
}
